encodings/datetime-parts/src/compute/compare.rs

```diff
@@ -0,0 +1,158 @@
+use vortex_array::array::ConstantArray;
+use vortex_array::compute::{and, compare, try_cast, CompareFn, Operator};
+use vortex_array::{Array, IntoArray};
+use vortex_datetime_dtype::TemporalMetadata;
+use vortex_dtype::DType;
+use vortex_error::VortexResult;
+
+use crate::array::{DateTimePartsArray, DateTimePartsEncoding};
+use crate::timestamp;
+
+impl CompareFn<DateTimePartsArray> for DateTimePartsEncoding {
+    /// Compares two arrays and returns a new boolean array with the result of the comparison.
+    /// Or, returns None if comparison is not supported.
+    ///
+    /// # NOTE: Only `Operator::Eq` is currently supported.
+    fn compare(
+        &self,
+        lhs: &DateTimePartsArray,
+        rhs: &Array,
+        operator: Operator,
+    ) -> VortexResult<Option<Array>> {
+        if operator != Operator::Eq {
+            return Ok(None);
+        }
+
+        let Some(rhs_const) = rhs.as_constant() else {
+            return Ok(None);
+        };
+        let Ok(Some(timestamp)) = rhs_const
+            .as_extension()
+            .storage()
+            .as_primitive()
+            .as_::<i64>()
+        else {
+            return Ok(None);
+        };
+
+        let DType::Extension(ext_dtype) = rhs_const.dtype() else {
+            return Ok(None);
+        };
+
+        let temporal_metadata = TemporalMetadata::try_from(ext_dtype.as_ref())?;
+        let ts_parts = timestamp::split(timestamp, temporal_metadata.time_unit())?;
+
+        let compare_dtp = |lhs: &Array, rhs: i64, operator: Operator| -> VortexResult<Array> {
+            match try_cast(ConstantArray::new(rhs, lhs.len()), lhs.dtype()) {
+                Ok(casted) => compare(lhs, casted, operator),
+                Err(_) => {
```

Comment:
Can you add a comment about what this error case means? i.e. the RHS is known to be larger than all LHS values.